### ISSUE TITLE
no-unused-declaration: doesn't work with types

### ISSUE THREAD

**Issue by Hotell** [cartant/eslint-plugin-etc#4]
There might be an issue with resolving declaration tokens via `import`

**Config:**

```
'etc/no-unused-declaration': ['error',{ignored:{'React': true}}]
```

**Example:**

✅No error

```ts
type Greeter = { who: string; greeting: string }

const greeter: Greeter = { who: 'John', greeting: 'Hello!' }

console.log(greeter)
```

🚨Error

```ts
// error: Unused declarations are forbidden (etc/no-unused-declaration)
import {Greeter} from './types';

const greeter: Greeter = { who: 'John', greeting: 'Hello!' }

console.log(greeter)
```

**Expected behaviour:**

- non of those should be error

**Comment by cartant** [comment]
Looks like a bug, to me.

**Comment by cartant** [comment]
It's actually a TODO: https://github.com/cartant/eslint-plugin-etc/blob/5ba4b87cd4600d6638b2d896324312313a1f58be/tests/rules/no-unused-declaration.ts#L174-L182

**Comment by cartant** [comment]
Should be fixed/implemented in `0.0.1-beta.12`.

There are some other things that are yet-to-be-done - unused import fixers, mostly - but they can be done later. Hopefully, with this fixed, the rule will be useful in its current form.
